cabot/cabotapp/alert.py

```diff
@@ -35,18 +37,33 @@ def __unicode__(self):
         return u'%s' % (self.title)
 
 
-def send_alert(service, duty_officers=[], fallback_officers=[]):
+def send_alert(service, duty_officers=None,
+               escalation_officers=None,
+               fallback_officers=None):
+
+    duty_officers = duty_officers or []
+    escalation_officers = escalation_officers or []
+    fallback_officers = fallback_officers or []
+
+    escalation_cutoff = timezone.now() - timedelta(
+        minutes=service.escalate_after)
+
     users = service.users_to_notify.filter(is_active=True)
+
     for alert in service.alerts.all():
-        try:
-            alert.send_alert(service, users, duty_officers)
-        except Exception:
-            logging.exception('Could not sent {} alert'.format(alert.name))
-            if fallback_officers:
-                try:
-                    alert.send_alert(service, users, fallback_officers)
-                except Exception:
-                    logging.exception('Could not send {} alert to fallback officer'.format(alert.name))
+        for user_list in [duty_officers, escalation_officers, fallback_officers]:
```

Comment:
Not sure I understand the logic here.  Seems like your new code:

1. Sends alerts to first non-empty officers list
2. If successful, quit
3. If an exception is thrown, check if we should escalate.  If we are scheduled to escalate, then loop and alert the next set of officers.  Else, quit.

Seems like if alerts are delivered successfully, we'll never escalate?  Or am I missing something?